server/service/src/programs/contact_trace/contact_trace_updated.rs

```diff
@@ -0,0 +1,61 @@
+use chrono::DateTime;
+use repository::{
+    contact_trace::{ContactTraceFilter, ContactTraceRepository},
+    contact_trace_row::{ContactTraceRow, ContactTraceRowRepository, ContactTraceStatus},
+    Document, ProgramRow, StorageConnection, StringFilter,
+};
+use util::hash::sha256;
+
+use super::{contact_trace_schema::SchemaContactTrace, upsert::UpsertContactTraceError};
+
+/// Callback called when a program enrolment document has been updated
+pub(crate) fn update_contact_trace_row(
+    con: &StorageConnection,
+    patient_id: &str,
+    document: &Document,
+    contact_trace: SchemaContactTrace,
+    program_row: ProgramRow,
+) -> Result<ContactTraceRow, UpsertContactTraceError> {
+    let datetime = DateTime::parse_from_rfc3339(&contact_trace.datetime)
+        .map_err(|err| {
+            UpsertContactTraceError::InternalError(format!(
+                "Invalid enrolment datetime format: {}",
+                err
+            ))
+        })?
+        .naive_utc();
+
+    // Retrieve existing or create now contact trace id
+    let repo = ContactTraceRepository::new(con);
+    let contact_trace_row = repo
+        .query_by_filter(ContactTraceFilter {
+            document_name: Some(StringFilter::equal_to(&document.name)),
+            ..ContactTraceFilter::default()
+        })?
+        .pop();
+    let contact_trace_id = match contact_trace_row {
+        Some(contact_trace_row) => contact_trace_row.0.id,
+        None => sha256(&document.name),
```

Comment:
Actually thought it would be nice to have stable row ids. For example, it the aux table is created on a fresh site sync you would get the same aux table row id. I did the same change for enrolments and encounter...

Reply:
I meant when populating the contact_trace table from the contact_trace document.
The name is only chosen once when creating the initial document, the name is not changing and thus the id is not changing. (please also see other comment about alternatives to choose the ID)

Reply:
But the uuid() will not change

Reply:
The uuid will change if you re-sync a site though. This isn't really a problem as long as user doesn't bookmarks the url to a specific contact tracing entry... Lets discuss these id questions in person think this would be quicker.